test/units/TEST-34-DYNAMICUSERMIGRATE.sh

```diff
@@ -148,13 +148,114 @@ EOF
     systemctl start testservice-34-check-writable.service
 }
 
+idmapped_mounts_avail() {
+    rm -rf /var/lib/testidmapped /var/lib/private/testidmapped
+
+    cat >/run/systemd/system/testservice-34-check-idmapped-avail.service <<\EOF
+[Unit]
+Description=Check id-mapped directories when DynamicUser=yes with StateDirectory
+
+[Service]
+Type=oneshot
+
+MountAPIVFS=yes
+DynamicUser=yes
+PrivateUsers=yes
+TemporaryFileSystem=/run /var/opt /var/lib /vol
+StateDirectory=testidmapped:sampleservice
+ExecStart=echo "hello"
+EOF
+
+    systemctl daemon-reload
+    systemctl start testservice-34-check-idmapped-avail.service
+    set +x
+    l=$(journalctl -b0 -u testservice-34-check-idmapped-avail.service)
```

Comment:
this is going to be racy - for a test it's fine to just check for a kernel version, check `test/units/TEST-62-RESTRICT-IFACES.sh` for an example